readthedocs/proxito/middleware.py

```diff
@@ -132,10 +132,30 @@ def process_request(self, request):  # noqa
         return None
 
     def process_response(self, request, response):  # noqa
-        """Set the Strict-Transport-Security (HSTS) header for a custom domain if max-age>0."""
-        if hasattr(request, 'domain'):
+        """
+        Set the Strict-Transport-Security (HSTS) header for docs sites.
+
+        * For the public domain, set the HSTS header if settings.PUBLIC_DOMAIN_USES_HTTPS
+        * For custom domains, check the HSTS values on the Domain object.
+          The domain object should be saved already in request.domain.
+        """
+        host = request.get_host().lower().split(':')[0]
+        public_domain = settings.PUBLIC_DOMAIN.lower().split(':')[0]
+
+        hsts_header_values = []
+
+        if not request.is_secure():
+            # Only set the HSTS header if the request is over HTTPS
+            return response
+
+        if settings.PUBLIC_DOMAIN_USES_HTTPS and (public_domain in host or host == 'proxito'):
```

Comment:
I don't think we want `proxito` here, since thats used locally and we don't have SSL there. 

```suggestion
        if settings.PUBLIC_DOMAIN_USES_HTTPS and public_domain in host:
```